lib/Onyx.js

```diff
@@ -981,32 +1008,27 @@ function applyMerge(key, data) {
  * Onyx.merge(ONYXKEYS.POLICY, {name: 'My Workspace'}); // -> {id: 1, name: 'My Workspace'}
  *
  * @param {String} key ONYXKEYS key
- * @param {(Object|Array)} value Object or Array value to merge
+ * @param {(Object|Array)} changes Object or Array value to merge
  * @returns {Promise}
  */
-function merge(key, value) {
-    if (mergeQueue[key]) {
-        mergeQueue[key].push(value);
-        return Promise.resolve();
-    }
-
-    mergeQueue[key] = [value];
-    return get(key)
-        .then((data) => {
+function merge(key, changes) {
+    const nextModification = () => get(key)
```

Comment:
I think I am right in thinking that this logic is "fine" to duplicate because concurrent calls to `get()` are already hooked into a single request?

Reply:
Instead of `queueModification`, we might have `waitPendingWrites(key)` which should make flow/logic more obvious

```
waitPendingWrites(key)
  .then(() => get(key))
```

Reply:
Agree, this naming would seem more obvious/easy